platform.txt

```diff
@@ -211,10 +214,15 @@ recipe.c.combine.pattern="{compiler.path}{compiler.c.elf.cmd}" "-Wl,--Map={build
 recipe.objcopy.partitions.bin.pattern={tools.gen_esp32part.cmd} -q "{build.path}/partitions.csv" "{build.path}/{build.project_name}.partitions.bin"
 
 ## Create bin
-recipe.objcopy.bin.pattern_args=--chip {build.mcu} elf2image --flash_mode "{build.flash_mode}" --flash_freq "{build.flash_freq}" --flash_size "{build.flash_size}" -o "{build.path}/{build.project_name}.bin" "{build.path}/{build.project_name}.elf"
+recipe.objcopy.bin.pattern_args=--chip {build.mcu} elf2image --flash_mode "{build.flash_mode}" --flash_freq "{build.flash_freq}" --flash_size "{build.flash_size}" --elf-sha256-offset 0xb0 -o "{build.path}/{build.project_name}.bin" "{build.path}/{build.project_name}.elf"
```

Comment:
`--elf-sha256-offset 0xb0` Is this change necessary? What does it do?

Reply:
Hi @me-no-dev the `sha256` of firmware is needed for debugging by ESP-Insights backend.

`esptool.py` from ESP-IDF inserts sha256 by default and is implicit. In case of Arduino however this change was not present. Maybe it was not found useful to add this at the point. 

https://github.com/espressif/esp-idf/blob/master/components/esptool_py/project_include.cmake#L48

Reply:
@ivankravets @valeros PTAL. This is needed for elf2bin part

Reply:
I guess it will suffice to add a hot fix directly to the PlatformIO build script in this repo until a new version of the core with this feature is released.

@sanketwadekar Please add the following snippet somewhere at the bottom of https://github.com/espressif/arduino-esp32/blob/master/tools/platformio-build.py:

```python
#
#  Adjust the `esptoolpy` command in the `ElfToBin` builder with firmware checksum offset
#

action = deepcopy(env["BUILDERS"]["ElfToBin"].action)
action.cmd_list = env["BUILDERS"]["ElfToBin"].action.cmd_list.replace(
    "-o", "--elf-sha256-offset 0xb0 -o"
)
env["BUILDERS"]["ElfToBin"].action = action
```
Also, an extra import should be added at the top of the same file:
```python
from copy import deepcopy
```
Thanks!

Reply:
PTAL